### PR TITLE
Made Navigator and PieChart Reusable/ Added Barchart to ReportsPage / Connected navigator with TopBar

### PR DESCRIPTION
**Pull request by Mihaidaniel22**
Bar Chart is not resizing, still needs work. The main part is the connection between date nav and day, week, month, year tabs.

### CODE REVIEW

File: src/components/common/customPieChart/CustomPieChart.jsx [hunk 1]
```diff
@@ -23,28 +17,25 @@ const CustomPieChart = () => {
...
-              cx: 200,
-            },
-          ]}
-          width={700}
-          height={300}
-        />
-      </Box>
+      <PieChart
+        sx={{
+          '--ChartsLegend-itemMarkSize': '20px',
```
Comment (en-teodora-velicu) [inline src/components/common/customPieChart/CustomPieChart.jsx:22]: this object can be a const outside the component

File: src/components/common/dateNavigation/DateNavigation.jsx [hunk 2]
```diff
@@ -1,28 +1,73 @@
...
 
-const DateNavigation = ({ selectedDate, onDateChange }) => {
   const isCurrentDay = isToday(selectedDate);
 
   const handlePrevDay = () => {
-    const newDate = subDays(selectedDate, 1);
-    onDateChange(newDate);
+    let newDate;
+
+    if (tab === 'Day') {
```
Comment (en-teodora-velicu) [inline src/components/common/dateNavigation/DateNavigation.jsx:44]: switch case is more fitted here.

File: src/components/common/dateNavigation/DateNavigation.jsx [hunk 3]
```diff
@@ -1,28 +1,73 @@
...
+
+    handleCalendarDateChange(newDate);
   };
 
   const handleNextDay = () => {
-    const newDate = addDays(selectedDate, 1);
-    onDateChange(newDate);
+    let newDate;
+
+    if (tab === 'Day') {
```
Comment (en-teodora-velicu) [inline src/components/common/dateNavigation/DateNavigation.jsx:60]: switch case here

File: src/components/common/dateNavigation/DateNavigation.jsx [hunk 4]
```diff
@@ -36,32 +81,95 @@ const DateNavigation = ({ selectedDate, onDateChange }) => {
...
-    onDateChange(newDate);
+    if (isAfter(newDate, new Date())) {
+      return;
+    }
+
+    setSelectedDate(newDate);
+
+    let start, end;
+
+    if (tab === 'Day') {
```
Comment (en-teodora-velicu) [inline src/components/common/dateNavigation/DateNavigation.jsx:95]: this function is too long. try to move this switch case in a different function. also these cases seem to be the same. try to create a utils function which can encapsulate the cases

File: src/components/common/dateNavigation/DateNavigation.jsx [hunk 5]
```diff
@@ -36,32 +81,95 @@ const DateNavigation = ({ selectedDate, onDateChange }) => {
...
   };
 
+  useEffect(() => {
+    console.log(dateRange);
+  }, [dateRange]);
+
+  useEffect(() => {
+    let newCalendarView = [];
+
+    if (tab === 'Day' || tab === 'Week') {
```
Comment (en-teodora-velicu) [inline src/components/common/dateNavigation/DateNavigation.jsx:139]: again here. substract in a different function

File: src/components/common/dateNavigation/DateNavigation.jsx [hunk 6]
```diff
@@ -36,32 +81,95 @@ const DateNavigation = ({ selectedDate, onDateChange }) => {
...
+    <Box className='navigation-container'>
+      <IconButton className='nav-button' onClick={handlePrevDay}>
         <KeyboardArrowLeftRoundedIcon />
       </IconButton>
-      <Box className="date-container">
+      <Box className='date-container'>
         <Typography>
-          {isCurrentDay ? (
-            <>Today - {format(selectedDate, "MMM d, yyyy")}</>
+          {tab === 'Day' && isCurrentDay ? (
```
Comment (en-teodora-velicu) [inline src/components/common/dateNavigation/DateNavigation.jsx:157]: do not nest ternary operations. its very hard to follow.

**Review by Alexandra18042003 — APPROVED** [review]

**Review by CristiStroescu — APPROVED** [review]
